### PR TITLE
Docs: Updated compile instructions for 3.12+

### PR DESCRIPTION
**Pull request by Simran-B**
The related docs PR removes the outdated compile instructions from 3.12+: https://github.com/arangodb/docs-hugo/pull/513

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -62,7 +62,22 @@ yet.
...
+    git clone --recurse-submodules --jobs 8 https://github.com/arangodb/arangodb
+
+This automatically clones the **devel** branch.
+
+If you only plan to compile a specific version of ArangoDB locally and do not
+want to commit changes, you can speed up cloning substantially by using the
+`--depth` and `--shallow-submodules` parameters and specifying a branch, tag,
+or commit hash using the `--branch` parameter for the clone command as follows:
+
+    git clone --depth 1 --shallow-submodules --branch v3.12.0 https://github.com/arangodb/arangodb
```
Comment (Simran-B) [inline CONTRIBUTING.md:78]:
```suggestion
    git clone --depth 1 --shallow-submodules --jobs 8 --branch v3.12.0 https://github.com/arangodb/arangodb
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -224,65 +239,86 @@ but do not include `Documentation/Metrics/allMetrics.yaml` in your PR
...
+version you want to build.
 
-```
--DTARGET_ARCHITECTURE
-```
+You can start the build container as follows:
 
-Oskar uses predefined architecture which is defined in `./VERSIONS` file or `sandybridge` if it's not defined.
+    cd arangodb
+    docker run -it -v $(pwd):/root/project arangodb/ubuntubuildarangodb-devel:3
```
Comment (Simran-B) [inline CONTRIBUTING.md:257]: `${pwd}` in PowerShell!

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -224,65 +239,86 @@ but do not include `Documentation/Metrics/allMetrics.yaml` in your PR
...
 
-    shellInAlpineContainer
+To build specific targets instead of all, you can specify them like shown below:
 
-    apk add --no-cache nodejs yarn && cd /work/ArangoDB/build && cmake --build . --target frontend
+    cmake --build --preset community --target arangod arangosh arangodump arangorestore arangoimport arangoexport arangobench arangovpack frontend
 
-To remove all available node modules and start a clean installation run:
+You can find the output files at `build-presets/<preset-name>/`,
+with the compiled executables in the `bin/` subfolder.
```
Comment (Simran-B) [inline CONTRIBUTING.md:285]: To collect the files and bring them into a release-like structure:
`cmake --install ./build-presets/community --prefix package`
(no --preset support, --prefix is the target folder, may not work if not all targets were built)

**Review by jsteemann — APPROVED** [review]
LGTM

**Review by mpoeter — APPROVED** [review]

**Pull request merged**